src/packet.c

```
@@ -3,295 +3,307 @@
+        return 1; /* Already seen in window -> replay */
+    }
+
+    tr->window_bits |= bit;
+    return 0; /* New within window */
+}
+
+/* ============================================================================
+ * Packet handling: forwarding and deduplication via recent_packets ring
+ * ============================================================================ */
+
+int packet_handle(const packet_t* packet, uint16_t from_node) {
```

Cppcheck (2.21.0)

[style] The function 'packet_handle' is never used.

(unusedFunction)